src/check_missing_alleles.py

```diff
@@ -0,0 +1,439 @@
+import csv
+import os
+from subprocess import Popen, PIPE
+
+
+def get_IMGT_data():
+    """Uses wget to retrieve the latest IMGT alleles + sequences from their github
+    These get deleted later after operations are complete
+    """
+    imgt_prot_URL = (
+        "https://raw.githubusercontent.com/ANHIG/IMGTHLA/Latest/hla_prot.fasta"
+    )
+    imgt_allele_URL = (
+        "https://raw.githubusercontent.com/ANHIG/IMGTHLA/Latest/Allelelist.txt"
+    )
+    process = Popen(["wget", imgt_prot_URL, imgt_allele_URL], stdout=PIPE, stderr=PIPE)
+    stdout, stderr = process.communicate()
+
+
+def update_chain_sequence():
+    """Updates the chain-sequence.tsv with missing alleles from IMGT
+
+    Returns:
+        A list of missing alleles ['A*02:01', 'B*02:01']
+    """
+
+    mro_alleles = set()
+    with open("../ontology/chain-sequence.tsv") as fh:
+        rows = csv.DictReader(fh, delimiter="\t")
+        for row in rows:
+            if "HLA" in row["Label"]:
+                mro_alleles.add(row["Label"].split(" ")[0][4:])
+
+    # Create a dictionary mapping the IMGT accession to protein sequence
+    seqs = {}
+    with open("./hla_prot.fasta") as fasta:
+        accession = None
+        seq = ""
+        for line in fasta:
+            if line.startswith(">"):
+                if accession:
+                    seqs[accession] = seq
+                accession = None
+                seq = ""
+
+                # Match the accession
+                if line.startswith(">HLA:HLA"):
+                    accession = line[5:13]
+                else:
+                    print("Bad accession:", line)
+            else:
+                seq += line.strip()
+        seqs[accession] = seq
+
+    imgt_dict = {}
+
+    # Get the mapping between allele name and IMGT accession
+    with open("./Allelelist.txt") as fh:
+        # Skip IMGT header
+        for _ in range(6):
+            next(fh)
+        rows = csv.DictReader(fh, delimiter=",")
+        for row in rows:
+            allele = row["Allele"]
+            allele = (":").join(allele.split(":")[:2])
+            if allele not in imgt_dict and "N" not in allele and "Q" not in allele:
+                imgt_dict[allele] = row["AlleleID"]
+
+    # Find differences between IMGT and MRO alleles
+    imgt_alleles = set(imgt_dict.keys())
+    missing_alleles = imgt_alleles.difference(mro_alleles)
+
+    missing_allele_rows = set()
+    for allele in missing_alleles:
+        try:
+            accession = imgt_dict[allele]
+            source = "IMGT/HLA"
+            label = f"HLA-{allele} chain"
+            resource_name = allele
+            seq = seqs[imgt_dict[allele]]
+            missing_allele_rows.add((label, resource_name, source, accession, seq))
+        except Exception:
+            continue
+
+    with open("../ontology/chain-sequence.tsv", "a+") as outfile:
+        for tup in missing_allele_rows:
+            outfile.write(("\t").join(tup) + "\n")
```

Comment:
`csv.writer(outfile, delimiter="\t", lineterminator="\n")` might be a better choice here